source/extensions/filters/http/ratelimit/ratelimit.cc

```diff
@@ -161,6 +171,21 @@ void Filter::onDestroy() {
   if (state_ == State::Calling) {
     state_ = State::Complete;
     client_->cancel();
+  } else {
+    // If the filter doesn't have an outstanding limit request (made during decodeHeaders) and has
+    // at least one policy with apply_on_stream_done=true, then we apply the rate limit here.
+    if (has_apply_on_stream_done_policy_) {
+      std::vector<Envoy::RateLimit::Descriptor> descriptors;
+      populateRateLimitDescriptors(descriptors, *request_headers_, true);
+      if (!descriptors.empty()) {
+        client_->cancel(); // Clears the internal state of the client, so that we can reuse it.
```

Comment:
Could you ensure this is OK?

Reply:
now i am working on the integration test and indeed this was unnecessary, so deleted!